### DISCUSSION TITLE
Updating tab width algo

### DISCUSSION THREAD

**Discussion by alaricljs** [kovidgoyal/kitty#5355] [General]
Due to my tab style preferences I am working on a different algorithm for `max_title_length` that distributes the modulo across some of the tabs.  This ends up making unequal tab sizes, but uses 95+% of the window width for the tabs.  My question is on the purpose of related code here:
https://github.com/kovidgoyal/kitty/blob/a487a7820077ecf71a3c48d9f35b8c273d444200/kitty/tab_bar.py#L591-L596

This branch doesn't normally execute for me, but the changes I described above do cause it to trigger occasionally.  When it does execute it is munging the tab in question which makes that tab not render at all.

What is the intent of the code?  I'd like to make it function appropriately if it makes sense.

**Comment by kovidgoyal** [comment]
Not sure what you are asking. The purpose of the code is when there is
not enough space left to draw more tab titles which can happen either
because the draw function didnt respect max_title_length or there are
simply too many tabs, the code will draw an ellipsis to indicate there
are more tabs that are not shown.

**Comment by alaricljs** [comment]
That is the answer to what I was asking.  In my testing with the 0.25.2 release of the code, the ellipsis does not get displayed until `self.screen.columns + 1` tabs have been created.  That doesn't seem to be terribly useful as the tabs themselves are just barely useful being merely ellipsis themselves at that point.  This is the case whether the intended content of the tab is large or small.

I'll try and think of an alternative to propose.
